frame/encode.go

```diff
@@ -0,0 +1,151 @@
+package frame
```

Comment:
adding this comment here but it's not related to this file

`frame/doc.go` file should be removed?

Reply:
I read on Rakyll's blog that a good place for package documentation is in a doc.go file. Google does this kind of thing, e.g.:

https://github.com/googleapis/google-cloud-go/blob/master/datastore/doc.go

This is why I created a doc.go file for each package with a brief explanation of the package contens.

Reply:
Ah I understand now. 

Very minor NIT: I suggest making the docs part a bit larger so it's obvious that it's a docs file (for those like me that are not familiar to that pattern).

This:

```
/* 

Package frame contains interfaces, types and functions to read and write CQL protocol frames 
as defined in sections 1 and 2 of the CQL protocol specifications.

*/
package frame

```
Instead of
```
// Package frame contains interfaces, types and functions to read and write CQL protocol frames as defined in sections 1 and 2 of the CQL protocol specifications.
package frame
```

Reply:
I actually had pretty much that initially, then I ran ` godoc -http=:6060` and realized that line breaks in the go file are rendered as line breaks too in the resulting HTML.

But this is also happening everywhere anyways, so I guess it's not a valid reason to avoid line breaks in the documentation we write. I will put those multi line comments back.